kafka-connector/src/main/java/com/google/pubsub/kafka/source/CloudPubSubSourceTask.java

```diff
@@ -169,40 +171,26 @@ public void start(Map<String, String> props) {
         Map<String,String> ack = Collections.singletonMap(cpsSubscription, ackId);
         SourceRecord record = null;
         if (hasCustomAttributes) {
-          SchemaBuilder valueSchemaBuilder = SchemaBuilder.struct().field(
-              ConnectorUtils.KAFKA_MESSAGE_CPS_BODY_FIELD,
-              Schema.BYTES_SCHEMA);
-
+          ConnectHeaders headers = new ConnectHeaders();
           for (Entry<String, String> attribute :
                messageAttributes.entrySet()) {
             if (!attribute.getKey().equals(kafkaMessageKeyAttribute)) {
-              valueSchemaBuilder.field(attribute.getKey(),
-                                       Schema.STRING_SCHEMA);
+              headers.addString(attribute.getKey(), attribute.getValue());
             }
           }
 
-          Schema valueSchema = valueSchemaBuilder.build();
```

Comment:
This represents a backwards-incompatible change. I think we should make this a configurable property so that users can choose how the attributes in a message from Cloud Pub/Sub are handled. When false (the default), the struct will be constructed. When true, headers can be used.

Reply:
I agree with you. It will be included in the next commit.